GitHub.Collectors.Functions/GitHubFunctions.cs

```diff
@@ -652,6 +652,97 @@ public async Task Traffic([QueueTrigger("traffic")] string queueItem, ExecutionC
+                ICache<ConditionalRequestTableEntity> requestsCache = new AzureTableCache<ConditionalRequestTableEntity>(telemetryClient, "requests");
+                await requestsCache.InitializeAsync().ConfigureAwait(false);
+                GitHubHttpClient httpClient = new GitHubHttpClient(this.httpClient, rateLimiter, requestsCache, telemetryClient);
+                IAuthentication authentication = this.configManager.GetAuthentication(CollectorType.Point, httpClient, repository.OrganizationLogin, this.apiDomain);
+
+                StorageManager storageManager;
+                List<IRecordWriter> recordWriters;
+                FunctionContextWriter<FunctionContext> contextWriter = new FunctionContextWriter<FunctionContext>();
+                using (storageManager = this.configManager.GetStorageManager(context.CollectorType, telemetryClient))
+                {
+                    recordWriters = storageManager.InitializeRecordWriters(identifier, context, contextWriter, this.adlsClient.AdlsClient);
+                    IRecordStatsTracker recordStatsTracker = null;
+
+                    foreach (IRecordWriter recordWriter in recordWriters)
+                    {
+                        recordWriter.SetOutputPathPrefix($"{repository.OrganizationId}/{repository.RepositoryId}");
+                        if (recordStatsTracker == null)
+                        {
+                            recordStatsTracker = recordWriter;
+                        }
+                    }
+
+                    statsTracker = new StatsTracker(telemetryClient, httpClient, recordStatsTracker, StatsTrackerRefreshFrequency);
+                    PointProcessor processor = new PointProcessor(authentication, recordWriters, httpClient, pointCache, telemetryClient, this.apiDomain);
+                    await processor.ProcessAsync(pointCollectorInput).ConfigureAwait(false);
+                }
+                await storageManager.FinalizeRecordWritersAsync().ConfigureAwait(false);
+                outputPaths = RecordWriterExtensions.GetOutputPaths(recordWriters);
+                success = true;
+            }
+            catch (GitHubRateLimitException exception)
+            {
+                CloudQueue trafficCloudQueue = await AzureHelpers.GetStorageQueueAsync("point").ConfigureAwait(false);
+                TimeSpan? initialVisibilityDelay = exception.getHiddenTime();
+                TimeSpan? timeToLive = null;
+                await trafficCloudQueue.AddMessageAsync(new CloudQueueMessage(queueItem), timeToLive, initialVisibilityDelay, new QueueRequestOptions(), new OperationContext()).ConfigureAwait(false);
+                telemetryClient.TrackException(exception, "RateLimiterRequeue");
+            }
+            catch (Exception exception)
+            {
+                telemetryClient.TrackException(exception);
+                throw exception;
```

Comment:
Alan kept fixing this for me, but using throw; instead of throw exception is the better approach. This way you don't generate another level of indirection in the stack trace :) See, Alan, I keep learning...